lib/enumerable/each_with_index.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each_with_index
+    raise 'Please provide a block' unless block_given?
+
```

Comment:
- [ ] my_each_with_index returns an enumerator if no block is given

Reply:
Added the cases you told me

Reply:
```ruby
- raise 'Please provide a block' unless block_given?
+ return to_enum unless block_given?
```
solved